full-stack-tests/presentation/src/frontend/UnifiedSelection.test.ts

```diff
@@ -185,6 +188,172 @@ describe("Unified Selection", () => {
       expect(imodel.selectionSet.has(instances.leafElement.key.id)).to.be.true;
     });
 
+    describe("Grouping Node", () => {
+      const addPartition = (classFullName: string, builder: TestIModelBuilder, name: string, parentId = IModel.rootSubjectId) => {
+        const parentProps: RelatedElementProps = {
+          relClassName: "BisCore:SubjectOwnsPartitionElements",
+          id: parentId,
+        };
+
+        const partitionProps: ElementProps = {
+          classFullName,
+          model: IModel.repositoryModelId,
+          parent: parentProps,
+          code: builder.createCode(parentId, BisCodeSpec.informationPartitionElement, name),
+        };
+        return builder.insertElement(partitionProps);
+      };
+
+      const addModel = (classFullName: string, builder: TestIModelBuilder, partitionId: string) => {
+        const modelProps: ModelProps = {
+          modeledElement: new RelatedElement({ id: partitionId }),
+          classFullName,
+          isPrivate: false,
+        };
+        return builder.insertModel(modelProps);
+      };
+
+      const addSpatialCategory = (builder: TestIModelBuilder, modelId: string, name: string, isPrivate?: boolean) => {
+        const spatialCategoryProps: CategoryProps = {
+          classFullName: "BisCore:SpatialCategory",
+          model: IModel.dictionaryId,
+          code: builder.createCode(modelId, BisCodeSpec.spatialCategory, name),
+          isPrivate,
+        };
+        return builder.insertElement(spatialCategoryProps);
+      };
+
+      const addPhysicalObject = (builder: TestIModelBuilder, modelId: string, categoryId: string, elemCode = Code.createEmpty()) => {
+
+        const physicalObjectProps: PhysicalElementProps = {
+          classFullName: "Generic:PhysicalObject",
+          model: modelId,
+          category: categoryId,
+          code: elemCode,
+        };
+        return builder.insertElement(physicalObjectProps);
+      };
+
+      const objectIdArray = Array<Id64String>();
+      let groupingNodesIModel: IModelConnection;
+      beforeEach(async () => {
+        groupingNodesIModel = await buildTestIModel("GroupByClass", (builder) => {
```

Comment:
I don't think sharing the iModel between all these tests is a good idea - that makes it contain stuff that's needed for all the tests rather than what's needed for each individual test. Instead, I think each test should create a minimal iModel itself. 

Also, make sure it's really **minimal** (at the moment you're inserting some models that don't seem to be relevant in any of the tests).

Ideally, the existing tests would also create their own iModel instead of using "Properties_60InstancesWithUrl2". I don't think we want to refactor them in this PR, but maybe just change them so that the iModel is opened in each test individually would be a good start that could be made here too. That would allow us to avoid having two iModels open at the same time (`imodel` & `groupingNodesIModel`)

Reply:
Removed a couple of objects from the imodels, but not sure if you had something more in mind?
Made each test create its own imodel.